### DISCUSSION TITLE
K-fold cross-validation using PyTorch Geometric

### DISCUSSION THREAD

**Discussion by bianchiniriccardo95** [pyg-team/pytorch_geometric#3662] [Q&A]
Hello!
First of all, thanks for this amazing tool. 
I have a question: 
I want to perform node classification on my network composed of 2410 nodes. Since now, I split my nodes using the function  `AddTrainValTestMask` function in order to validate the machine, but my idea is to perform a **K-fold cross validation** of the network, to gain better confidence about the goodness of the GCN.
How can I do this? How can I split my nodes in the different masks multiple times? Is there a specific function?
Thanks in advance.

**Comment by rusty1s** [comment]
Such a function doesn't exist yet, I'm sorry. I suggest to use `sklearn.model_selection.StratifiedKFold` for this. [Here](https://github.com/pyg-team/pytorch_geometric/blob/master/benchmark/kernel/train_eval.py#L82-L97) is an example of it within PyG.

**Reply by bianchiniriccardo95** [reply]
Thank you very much for your help!

**Reply by WhItEG-bit** [reply]
Thanks，help a lot!
